cxx/distributions/bigram.hh

```diff
@@ -73,11 +76,15 @@ class Bigram : public Distribution<std::string> {
     }
     return total_logp;
   }
+
   double logp_score() const {
-    return std::transform_reduce(
-        transition_dists.cbegin(), transition_dists.cend(), 0, std::plus{},
-        [&](auto d) -> double { return d.logp_score(); });
+    double logp = 0;
+    for (auto d : transition_dists) {
```

Comment:
Readability but also avoiding errors. The bug in the previous version was using `0` instead of `0.0` (because of transform_reduce using the type of the initial value, and casting implicitly). Thus we were getting integers and getting the wrong log prob.